fastavro/reader.py

```diff
@@ -375,10 +471,17 @@ def read_data(fo, writer_schema, reader_schema=None):
     """Read data from file object according to schema."""
 
     record_type = extract_record_type(writer_schema)
+    logical_type = extract_logical_type(writer_schema)
```

Comment:
No need for new variable. IMO change `extract_record_type` to return the logical type.